widget_driver/lib/src/widget_driver_provider.dart

```diff
@@ -13,4 +13,10 @@ abstract class WidgetDriverProvider<Driver extends WidgetDriver> {
 
   /// Creates and returns the `Driver` with the hard coded test values.
   Driver buildTestDriver();
+
+  /// This function only get's overriden when the driver has provided properties.
+  /// It calls a function in the driver, added by the `[SomeDriver]ProvidedPropertiesMixin`,
+  /// to respond to properties changed during a state update.
+  /// (The driver does not get rebuilt for state updates!)
+  void updateDriverProvidedProperties(Driver driver) {}
```

Comment:
Maybe mention here that this gets called everytime the widgets build method is called.. right before the build method..